aws-opentelemetry-distro/src/amazon/opentelemetry/distro/_aws_metric_attribute_generator.py

```diff
@@ -674,6 +703,149 @@ def _set_span_kind_for_dependency(span: ReadableSpan, attributes: BoundedAttribu
     attributes[AWS_SPAN_KIND] = span_kind
 
 
+def _get_bedrock_agentcore_resource_type_and_identifier(
+    span: ReadableSpan,
+) -> tuple[Optional[str], Optional[str], Optional[str]]:
+    """Get BedrockAgentCore resource type, identifier, and CFN primary identifier based on span attributes."""
+    attrs = span.attributes
+    if not attrs:
+        return None, None, None
+
+    def format_resource_type(resource_type: Optional[str]) -> Optional[str]:
+        return f"{_NORMALIZED_BEDROCK_AGENTCORE_SERVICE_NAME}::{resource_type}" if resource_type else None
+
+    for handler in [
+        _handle_browser_attrs,
+        _handle_gateway_attrs,
+        _handle_runtime_attrs,
+        _handle_code_interpreter_attrs,
+        _handle_identity_attrs,
+        _handle_memory_attrs,
+    ]:
+        resource_type, resource_identifier, cfn_primary_identifier = handler(attrs)
+        if resource_type:
+            return format_resource_type(resource_type), resource_identifier, cfn_primary_identifier
+
+    return None, None, None
+
+
+def _handle_browser_attrs(attrs) -> tuple[Optional[str], Optional[str], Optional[str]]:
+    browser_id = attrs.get(GEN_AI_BROWSER_ID)
+    browser_arn = attrs.get(AWS_BEDROCK_AGENTCORE_BROWSER_ARN)
+    if browser_id or browser_arn:
+        resource_identifier = None
+        if browser_arn:
+            resource_identifier = extract_bedrock_agentcore_resource_id_from_arn(str(browser_arn))
+        if browser_id:
+            resource_identifier = str(browser_id)
```

Comment:
Do you always need to check on two if `browser_id` value exists?
```suggestion
        if browser_id:
            resource_identifier = str(browser_id)
        elif browser_arn:
            resource_identifier = extract_bedrock_agentcore_resource_id_from_arn(str(browser_arn))
```

Reply:
ack. That makes sense.